### ISSUE TITLE
Two ways in which the default for mpi_memory_alloc_kinds can be interpreted

### ISSUE THREAD

**Issue by th-hlrs** [mpi-forum/mpi-issues#773]
# Problem

The new `mpi_memory_alloc_kinds` can be used to request and to query support for different. If it is used to query what the default is, current wording ('string, default: "mpi,system"') can be read as

1.  a conforming library has to exactly set to/return the value "mpi,system" if nothing else has been requested (e.g. by an start-up mechanism). Thus, no library could by default enable support for GPUs, even if it would make sense;
2. a conforming library sets the info key/returns the value as if "mpi,system" had been specified by an start-up mechanism. Then also additional memory kinds could be added.

# Proposal

It might not make sense to restrict libraries to only offer (default) support for "mpi,system". Then 2. would be better.

# Changes to the Text

Add in section 11.4.3 in the definition of `mpi_memory_alloc_kinds` text similar to:
The default is treated as if the user had requested it.

# References and Pull Requests

**Comment by hppritcha** [comment]
although this is in the process creation chapter, i've not been very involved in the memory kind allocation portion.  reassigning to @jdinan

**Comment by jdinan** [comment]
It was intended that libraries can include implementation-defined kinds in addition to the default. This is specified further down in the section, but I do agree it needs to be clarified here. Here's one way to clarify this:
> "mpi_memory_alloc_kinds" (string, default: "mpi,system"):  A comma separated list of memory allocation kinds. ++The default must contain the kinds specified and may contain additional implementation-defined kinds.++

**Comment by wesbland** [comment]
No-no commits: https://github.com/mpi-forum/mpi-standard/pull/986/commits/67b9ea749a50e10bec3e34e261f61e9a21b4e236

**Comment by wesbland** [comment]
This passed a [no-no vote](https://mpi-forum.org/meetings/2025/01/votes).
<table>
<tr>
    <td> Yes
    <td> No
    <td> Abstain
<tr>
    <td> 30
    <td> 0
    <td> 0
</table>

**Comment by wesbland** [comment]
This passed a [errata vote](https://mpi-forum.org/meetings/2025/01/votes).
<table>
<tr>
    <td> Yes
    <td> No
    <td> Abstain
<tr>
    <td> 30
    <td> 0
    <td> 0
</table>
